inc/input_queue.hpp

```diff
@@ -3,38 +3,68 @@
 
 namespace dragonslave {
 
-
 enum class InputType {
     KEYBOARD,
     MOUSE_POS,
     MOUSE_BUTTON,
     MOUSE_SCROLL,
 };
 
-
-struct InputEvent 
+struct MouseEvent
```

Comment:
Can we do something like `InputMousePressEvent`? It's a little verbose, but I think it'll help with name collisions.